packages/rest/src/controllers/didcomm/connections/ConnectionsController.ts

```diff
@@ -0,0 +1,137 @@
+import type { DidCommConnectionsRecord } from './ConnectionsControllerTypes'
+
+import { DidExchangeState, Agent, RecordNotFoundError } from '@credo-ts/core'
+import { Controller, Delete, Example, Get, Path, Post, Query, Request, Route, Security, Tags } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import { RequestWithAgent } from '../../../authentication'
+import { apiErrorResponse } from '../../../utils/response'
+import { Did } from '../../did/DidsControllerTypes'
+import { RecordId } from '../../types'
+
+import { connectionsRecordExample } from './ConnectionsControllerExamples'
+import { connectionRecordToApiModel } from './ConnectionsControllerTypes'
+
+@Tags('DIDComm Connections')
+@Route('/didcomm/connections')
+@Security('tenants', ['tenant'])
+@injectable()
+export class ConnectionsController extends Controller {
+  public constructor(private agent: Agent) {
+    super()
+  }
+
+  /**
+   * Find connection record by query
+   */
+  @Example<DidCommConnectionsRecord[]>([connectionsRecordExample])
+  @Get('/')
+  public async findConnectionsByQuery(
+    @Request() request: RequestWithAgent,
+    @Query('outOfBandId') outOfBandId?: RecordId,
+    @Query('alias') alias?: string,
+    @Query('state') state?: DidExchangeState,
+    @Query('did') did?: Did,
+    @Query('theirDid') theirDid?: Did,
+    @Query('theirLabel') theirLabel?: string,
+  ) {
+    const connections = await request.user.agent.connections.findAllByQuery({
+      alias,
+      did,
+      theirDid,
+      theirLabel,
+      state,
+      outOfBandId,
+    })
+
+    return connections.map(connectionRecordToApiModel)
+  }
+
+  /**
+   * Retrieve connection record by connection id
+   * @param connectionId Connection identifier
+   * @returns ConnectionRecord
+   */
+  @Example<DidCommConnectionsRecord>(connectionsRecordExample)
+  @Get('/:connectionId')
+  public async getConnectionById(@Request() request: RequestWithAgent, @Path('connectionId') connectionId: RecordId) {
+    const connection = await request.user.agent.connections.findById(connectionId)
+
+    if (!connection) {
+      this.setStatus(404)
+      return apiErrorResponse(`connection with connection id "${connectionId}" not found.`)
+    }
+
+    return connectionRecordToApiModel(connection)
+  }
+
+  /**
+   * Deletes a connection record from the connection repository.
+   *
+   * @param connectionId Connection identifier
+   */
+  @Delete('/:connectionId')
```

Comment:
Ah I commented too early... I guess with the Delete method it makes sense to separate it to stay conistent method-wise.